### PR TITLE
Use `UserPuckStyleKit` to draw `UserPuckStyleKitView`.

### PR DESCRIPTION
**Pull request by ShanMa1991**
### Description
This pr is to export the `UserPuckStyleKit` from the current and original resource of the navigation puck [design](https://github.com/mapbox/navigation-ui-resources/blob/main/Navigation%20Puck.pcvd), and use it to draw the `UserPuckStyleKitView`. It will be aligned with other views including `ManeuversStyleKit`, `SpeedLimitStyleKit` and `LanesStyleKit`. And it also supports the automatic resizing to fix the #3444 as well.

### CODE REVIEW

File: Sources/MapboxNavigation/UserPuckStyleKit.swift [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+//  Created on 2017-10-11.
+//  Copyright © 2017 Mapbox. All rights reserved.
+//
+//  Generated by PaintCode
+//  http://www.paintcodeapp.com
+//
+
+import UIKit
+
+public class UserPuckStyleKit : NSObject {
```
Comment (MaximAlien) [inline Sources/MapboxNavigation/UserPuckStyleKit.swift:14]: Should it be public?

**Review by MaximAlien — APPROVED** [review]
In general looks good to me. Be sure to test various edge-cases (like custom size of `UserCourseView`) to verify that there are no regressions.

File: Sources/MapboxNavigation/UserPuckStyleKit.swift (hunk 1 above)
Comment (ShanMa1991) [reply Sources/MapboxNavigation/UserPuckStyleKit.swift:14]: The `ManeuversStyleKit`, `SpeedLimitStyleKit` and `LanesStyleKit` are all public now. But because the `UserPuckStyleKit` right now is only used to draw the `UserPuckCourseView`. And we could customize it through the  `UserPuckCourseView`, I'll remove its public exposure.
